pyeudiw/jwt/__init__.py

```diff
@@ -38,23 +46,51 @@
     "EC": "A256GCM"
 }
 
+KeyLike = ECKey | RSAKey | OKPKey | SYMKey
+SerializationFormat = Literal["compact", "json"]
 
-class JWEHelper():
-    """
-    The helper class for work with JWEs.
-    """
 
-    def __init__(self, jwk: Union[JWK, dict]):
+logger = logging.getLogger(__name__)
+
+
+class JWHelperInterface:
+    def __init__(self, jwks: list[KeyLike | dict] | KeyLike | dict):
         """
         Creates an instance of JWEHelper.
 
-        :param jwk: The JWK used to crypt and encrypt the content of JWE.
-        :type jwk: JWK
+        :param jwks: The list of JWK used to crypt and encrypt the content of JWE.
+
         """
-        self.jwk = jwk
-        if isinstance(jwk, dict):
-            self.jwk = JWK(jwk)
-        self.alg = DEFAULT_SIG_KTY_MAP[self.jwk.key.kty]
+        if isinstance(jwks, dict):
+            single_jwk = key_from_jwk_dict(jwks)
+            single_jwk.add_kid()
```

Comment:
L'operazione `add_kid` muta il kid di chiavi che potenzialmente leggiamo dal mondo esterno e questo può a portare bug se qualcun altro utilizza dei kid diversi da jwk thumbprint con sha-256. Usare i thumbprint per i kid è SHOULD ma non mandatory MUST da rfc 7517, [vedi specs](https://datatracker.ietf.org/doc/html/rfc7517#section-4.5)

Esempi di questo scenario:
1. Qualcuno pubblica una chiave con {"kid" : "non-standard"}, noi la leggiamo come chiave con {"kid": "OUNx…95m-Fw"} e non siamo in grado di verificare un token jws che ha nel suo header contiene il {"kid": "non-standard"} perché non riusciamo a trovare la chiave che ha firmato, avendo sovrascritto il kid (questo non è uno scenario ipotetico, vedi ad esempio [questo membro della federazione CIE](https://federa.lepida.it/gw/OidcSaFull/.well-known/openid-federation) con kid non standard)
2. Qualcuno pubblica una chiave di cifratura RSA con {"kid" : "non-standard"}, noi la leggiamo come chiave con {"kid": "OUNx…95m-Fw"} e gli diamo un token jwe cifrato con header non protetto {"kid": "OUNx…95m-Fw"}: il destinatario non sa come decifrare il token perché io ho inserito un kid _diverso_ da quello della chiave che lui mi ha consegnato.

Reply:
@LadyCodesItBetter `add_kid` va eseguito solo se `kid` non fosse presente

Reply:
Dato che questo è direttamente il materiale JWK fornito da utilizzare, ed JWK è normato da RFC7517, dovremmo assumere che il kid potrebbe non essere presente come indicato in questa sezione della [reference spec](https://datatracker.ietf.org/doc/html/rfc7517#section-4.5)